src/main/java/com/salesforce/storm/spout/dynamic/kafka/Consumer.java

```diff
@@ -442,98 +457,43 @@ private void fillBuffer() {
-
-                    final long endOffset = getKafkaConsumer().position(assignedTopicPartition);
-
-                    logger.warn(
-                        "KAFKA SEEK - On {} Seeking {} (lastPersistedOffset = {}, lastStartedOffset = {}, endOffset = {})",
-                        assignedTopicPartition,
-                        resetOffset,
-                        lastPersistedOffset,
-                        lastStartedOffset,
-                        endOffset
-                    );
-
-                    getKafkaConsumer().seek(assignedTopicPartition, resetOffset);
-                } else {
-                    resetPartitions.add(assignedTopicPartition);
-                }
-
-                continue;
-            }
-
-            // This partition did NOT have any errors, but its possible that we "lost" some messages
-            // during the poll() call.  This partition needs to seek back to its previous position
-            // before the exception was thrown.
-
-            // Annoyingly we can't ask the KafkaConsumer for the current position via position() because
-            // it will assume we've consumed the messages it received prior to throwing the exception
-            // and we'll skip message, and we have no way to access those already consumed messages.
-
-            // Whatever offset is returned here, we've already played into the system.  We could try to seek
-            // to the this offset + 1, but there's no promise that offset actually exists!  If we did that and it doesn't
-            // exist, we'll reset that partition back to the earliest erroneously.
-
-            // The short of this means, we're going to replay a message from each partition we seek back to, but
-            // thats better than missing offsets entirely.
+                // Seek to the end we found above.  The end may have moved since we last asked, which is why we are not doing seekToEnd()
+                getKafkaConsumer().seek(
+                    topicPartition,
+                    endingOffset
+                );
 
-            // If offset is -1
-            if (lastStartedOffset == -1) {
-                // That means we haven't started tracking any offsets yet, we should seek to earliest on this partition
-                logger.info("Partition {} has no stored offset, resetting to earliest {}", assignedConsumerPartition, lastStartedOffset);
-                resetPartitionsToEarliest(Collections.singletonList(assignedTopicPartition));
+                partitionOffsetsManager.replaceEntry(
+                    new ConsumerPartition(topicPartition.topic(), topicPartition.partition()),
+                    endingOffset
+                );
             } else {
-                logger.info("Backtracking {} offset to {}", assignedConsumerPartition, lastStartedOffset);
-                getKafkaConsumer().seek(assignedTopicPartition, lastStartedOffset);
```

Comment:
Lets add a comment here that these ones are being seeked to head or tail or whatever we decide

Reply:
It looks like we call seekToEarliest() is that what we wanna do? Or seek to tail?